### PR TITLE
Add udunits xml path to site.cfg

### PR DESCRIPTION
**Pull request by dkillick**
Added path to UDUNITS XML database in cf_units' `site.cfg` in the build scripts. This is necessary as the location of this XML file varies between Win and not-Win:

* *nix: `$PREFIX/share/udunits/udunits2.xml`
* Win: `%PREFIX%/Library/share/udunits/udunits2.xml`

Note that on Win, `%PREFIX%/Library` can be accessed with the single environment variable "`%LIBRARY_PREFIX%`".

Note: relies on a change to cf_units to pick the change in XML database location, which is referenced below.

### CODE REVIEW

File: cf_units/build.sh [hunk 1]
```diff
@@ -9,7 +9,8 @@ fi
 
 SITECFG=cf_units/etc/site.cfg
 echo "[System]" > $SITECFG
-echo "udunits2_path = $PREFIX/lib/libudunits2.${EXT}" >> $SITECFG
```
Comment (dkillick) [inline cf_units/build.sh]: Raagh, stupid Windows. The difference between these two line is a Win-CRLF (I didn't set the git config option `autocrlf` until _after_ I pushed this :unamused:).

**Comment by bjlittle** [comment]
Bump the version or build number ...

**Comment by dkillick** [comment]
Right, there are two (two!) different errors causing this to fail on CI:

* **OSX, Linux**: error in conda-build: `Exception: info/files: duplicates`
* **Windows**: `OSError: [UT_OPEN_ARG] Failed to open UDUNITS-2 XML unit database : "No such file or directory"`

**Comment by bjlittle** [comment]
@dkillick Would cf_units v1.1.1 fix you're windows issue? You could update the recipe to point to the latest cf_units tag and it might resolve this ...

Not sure about the first issue, you can ssh into the circle-ci vm to poke around ... might help diagnose.

**Comment by dkillick** [comment]
@bjlittle it should do - v1.1.1 includes the related change that I made to cf_units to change the XML path. You never know, it might even solve the other problem!

**Comment by bjlittle** [comment]
@dkillick Agreed. Push it as a separate PR to this and we'll get it in, then rebase ...

**Comment by dkillick** [comment]
Bump version of cf_units.

**Comment by dkillick** [comment]
Oh phew, that's got AppVeyor and Windows playing ball - now it's just Travis and Circle to wrestle into submission.

And it's not often that one can say that Windows is the most amenable platform :flushed:

**Comment by dkillick** [comment]
Thanks to the tight coupling between this and #172 I've actually ended up bumping the version of cf_units to v1.1.1 in *this* PR. I'll close #172 in light of this fact.

**Comment by dkillick** [comment]
Hurrah! All the tests are passing at last :grinning: 

Ping @bjlittle @marqh @pelson - it would be good to get this in.

**Comment by raghu330** [comment]
Hi there! 
I am using conda installation of iris on windows and use git-bash. I get this strange error when I try to import iris with "OSError: [UT_PARSE] Failed to open UDUNITS-2 XML unit database : "No such file or directory"" in my G: drive, while everything is smooth in my C: drive. I believe that user 'pearseb' too reported this some 6-7 months ago. Any fix for this?

**Comment by dkillick** [comment]
Hi @raghu330 - what you report is a completely separate problem to those fixed by this PR. This PR ensured that the filepath for the UDUNITS2 XML database was correct on Windows installs.

The failure you're reporting is that, as it stands, UDUNITS2 used on Windows is demonstrating a bug that, at its simplest, means it cannot load its XML database if the Python prompt is started from a Windows drive that is not the same drive as where you have conda installed cf-units to. This is a UDUNITS2 bug that has been raised with UDUNITS here: Unidata/UDUNITS-2#43. Unfortunately, there's no fix for this for us until UDUNITS2 can fix it within their code.

In your case then, I expect you have installed to your C: drive. The simple workaround to this is to only ever execute the Python prompt from your C: drive, and other more complicated work-arounds, such as changing drive letters within Python before you import Iris, may also be possible. See also: SciTools/conda-recipes-scitools#115.

**Comment by raghu330** [comment]
Thanks @dkillick, I will go head and change working drives within the code for the time being. Raghu

**Pull request merged**